src/cmd/package.go

```diff
@@ -128,18 +129,47 @@ var packageMirrorCmd = &cobra.Command{
 		}
 	},
 	RunE: func(cmd *cobra.Command, args []string) error {
-		packageSource, err := choosePackage(args)
+		var c *cluster.Cluster
+		if dns.IsServiceURL(pkgConfig.InitOpts.RegistryInfo.Address) || dns.IsServiceURL(pkgConfig.InitOpts.GitServer.Address) {
```

Comment:
The way this code is right now, if someone has an internal git registry, and an external git server or vice versa mirror resources will break. Maybe it would be better to split apart this command into two sub commands that do one or the other. 

We are breaking most of the existing use cases by not accepting localhost for the in cluster registry service. If we're committed to doing that, it might be a clearer signal to users to just change the command in general.

Reply:
We can add an additional check before if that makes things better. I dont know if we have such a use case but it would be simple to support.

In regards to the breaking change I am bringing things inline with what the git server configuration is. Something that should have been done from the beginning. In the current solution we have two different logic for the git server and registry.  Currently any localhost address for the registry is assumed to be an internal registry, which is a large assumption to make. I have updated the docs and we can add a note in the Changelog, but we have to be able to make breaking changes that adjust behavior to be more standardized, especially before a v1 release.

Reply:
Gotcha, yeah an additional check to support would be good. I believe software factory might have that use case, though I'm not 100% sure. They use the Zarf registry, but instead of the Zarf gitea instance they spin up their own gitlab instance.